### PR TITLE
8299600: Use Objects.check*() where appropriate in java.io

### PR DESCRIPTION
**Pull request by stsypanov**
Use the following methods instead of hand-written code:
- Objects.checkFromIndexSize()
- Objects.checkFromToIndex()

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8299600](https://bugs.openjdk.org/browse/JDK-8299600): Use Objects.check*() where appropriate in java.io

### Reviewers
 * [Alan Bateman](https://openjdk.org/census#alanb) (@AlanBateman - **Reviewer**)
 * [Brian Burkhalter](https://openjdk.org/census#bpb) (@bplb - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/11849/head:pull/11849` \
`$ git checkout pull/11849`

Update a local copy of the PR: \
`$ git checkout pull/11849` \
`$ git pull https://git.openjdk.org/jdk pull/11849/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 11849`

View PR using the GUI difftool: \
`$ git pr show -t 11849`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/11849.diff">https://git.openjdk.org/jdk/pull/11849.diff</a>

</details>

### CODE REVIEW

File: src/java.base/share/classes/java/io/CharArrayWriter.java [hunk 1]
```diff
@@ -97,10 +98,8 @@ public void write(int c) {
      *          of the given array
      */
     public void write(char[] c, int off, int len) {
-        if ((off < 0) || (off > c.length) || (len < 0) ||
-            ((off + len) > c.length) || ((off + len) < 0)) {
-            throw new IndexOutOfBoundsException();
-        } else if (len == 0) {
+        Objects.checkFromToIndex(off, len, c.length);
```
Comment (swpalmer) [inline src/java.base/share/classes/java/io/CharArrayWriter.java:101]: shouldn't that be Objects.checkFromToIndex(off, off+len, c.length);

Comment (bplb) [reply src/java.base/share/classes/java/io/CharArrayWriter.java:101]: It looks like all the occurrences of `checkFromToIndex()` should instead be `checkFromIndexSize()`.

Comment (bplb) [reply src/java.base/share/classes/java/io/CharArrayWriter.java:101]: Indeed there are at least four regression test failures with this as-is:

- java/io/InputStream/ReadParams.java
- java/io/OutputStream/WriteParams.java
- java/io/Writer/WriteParams.java
- java/nio/channels/FileChannel/Transfer2GPlus.java

Comment (AlanBateman) [reply src/java.base/share/classes/java/io/CharArrayWriter.java:101]: > Indeed there are at least four regression test failures with this as-is:
> 
> * java/io/InputStream/ReadParams.java
> * java/io/OutputStream/WriteParams.java
> * java/io/Writer/WriteParams.java
> * java/nio/channels/FileChannel/Transfer2GPlus.java

The java/io tests run in tier2 and I think the GHA only run tier1.

Comment (stsypanov) [reply src/java.base/share/classes/java/io/CharArrayWriter.java]: Fixed along with copyright year

Comment (bplb) [reply src/java.base/share/classes/java/io/CharArrayWriter.java]: The regression tests now pass.

**Review by AlanBateman — APPROVED** [review]
Version 8af035f5 looks right and I see that @bplb has run the tests.

**Review by bplb — APPROVED** [review]

**Pull request closed without merging**